src/screens/Profile/Notification.js

```diff
@@ -0,0 +1,138 @@
+import React from 'react';
+import { View, StyleSheet, AsyncStorage } from 'react-native';
+import { Card, Button, Paragraph } from 'react-native-paper';
+import auth from '@react-native-firebase/auth';
+import Swipeable from 'react-native-gesture-handler/Swipeable';
+import PropTypes from 'prop-types';
+
+export default class Notification extends React.Component {
+  constructor(props) {
+    super(props);
+    const { viewed } = this.props;
+    this.state = { viewed, visible: true };
+    this.onClearPress = this.onClearPress.bind(this);
+    this.onViewPress = this.onViewPress.bind(this);
+  }
+
+  componentDidMount() {
+    this.unsubscribeFromAuth = auth().onAuthStateChanged((user) => {
+      if (user) {
+        this.setState({ currentUserID: user.uid });
+      }
+    });
+  }
+
+  async onViewPress() {
+    const { notificationID } = this.props;
+    const { currentUserID } = this.state;
+    await AsyncStorage.getItem(currentUserID).then((value) => {
+      const notificationArray = JSON.parse(value);
+      const notif = notificationArray.find((el) => el.id === notificationID);
+      if (notif.viewed === false) {
+        const updatedArray = notificationArray.filter((val) => val.id !== notificationID);
+        notif.viewed = true;
+        updatedArray.push(notif);
+        AsyncStorage.setItem(currentUserID, JSON.stringify(updatedArray));
+        this.setState({ viewed: true });
+      }
+    });
+  }
+
+  async onClearPress() {
+    const { notificationID } = this.props;
+    const { currentUserID } = this.state;
+    await AsyncStorage.getItem(currentUserID).then(async (value) => {
+      if (value) {
+        const notificationArray = JSON.parse(value);
+        const updatedArray = notificationArray.filter((val) => val.id !== notificationID);
+        await AsyncStorage.setItem(currentUserID, JSON.stringify(updatedArray));
+        this.setState({ visible: false });
+      }
+    });
+  }
+
+  render() {
+    const { message, type, replies } = this.props;
+    const { viewed, visible } = this.state;
+    const active = viewed ? notificationStyles.inactiveNotification
+      : notificationStyles.activeNotification;
+    const replyText = ` has received ${replies} replies. Click to go to post`;
+    const approvedText = ' has been approved. Click to go to post';
+    let text;
+    if (type === 'reply') {
+      text = replyText;
+    } else if (type === 'approve') {
+      text = approvedText;
+    }
+    const RightActions = () => (
+      <View style={notificationStyles.rightSwipeable}>
+        <Button onPress={this.onClearPress} style={{ backgroundColor: '#ffffff', marginLeft: 5 }}>Clear</Button>
+      </View>
+    );
+    return (
+      visible && (
```

Comment:
Notifications get deleted when you clear them (`onClearAsyncNotifications` calls `setState(notifications: [])`, so there's no reason to a have a `visible` state.

Reply:
For me: visible is for onClearPress(), because it does not recall componentDidMount in ForumNotificationScreen when onClearPress() for the Notification is called, so we need to set the state of visible to be false in the meantime and make the notification un-vieweable client side

Reply:
If you made the change above (lines 44 to 51) where you put a handler in the parent, I don't think you will need this `visible` state anymore to achieve this.